LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDeviceClient.cs

```diff
@@ -155,12 +154,165 @@ public async Task<bool> SendEventAsync(LoRaDeviceTelemetry telemetry, Dictionary
             return false;
         }
 
+        readonly SemaphoreSlim receiveAsyncLock = new SemaphoreSlim(1, 1);
```

Comment:
nit: can you make those explicitly private and move them to the other declarations?